src/ic-websocket-cdk/src/state.rs

```diff
@@ -79,16 +80,22 @@ pub(crate) fn increment_gateway_clients_count(gateway_principal: GatewayPrincipa
     });
 }
 
-/// Decrements the clients connected count for the given gateway.
-/// If there are no more clients connected, the gateway is removed from the list of registered gateways.
-pub(crate) fn decrement_gateway_clients_count(gateway_principal: &GatewayPrincipal) {
+/// Decrements the clients connected count for the given gateway, if it exists.
+///
+/// If `remove_if_empty` is true, the gateway is removed from the list of registered gateways
```

Comment:
Is `remove_if_empty` a parameter that you can pass to decide whether to remove the gateways? If so, why would one set it to `false`?

Reply:
See https://github.com/omnia-network/ic-websocket-cdk-rs/pull/5#discussion_r1427701673